src/Cello/problem_InitialValue.cpp

```diff
@@ -157,48 +91,47 @@ void InitialValue::enforce_block ( Block * block,
 
     } else if (parameter_type == parameter_list) {
 
-      // Check parameter length
+      Value value(parameters_, field_name);
```

Comment:
The `MaskPng` objects are re-initialized each time the `Value` constructor is called, which here is once per block.  For a problem involving lots of blocks (e.g. the load balancing test problem `input/load-balance.in`, which generates over 5000 blocks at initialization) it's re-reading the PNG file lots of times, leading to a performance issue. `InitialValue` originally initialized the masks at the start and saved them; you may want to try something similar with the `Value` objects.